pkg/task/manager.go

```diff
@@ -59,24 +66,89 @@ func (t *Task) IsExpired() bool {
 	return false
 }
 
+// Init manager supported by boltdb.
+func (m *manager) Init(workDir string) error {
+	bdb, err := bolt.Open(filepath.Join(workDir, "task.db"), 0655, nil)
+	if err != nil {
+		return errors.Wrap(err, "create task database")
+	}
+	m.db = bdb
+	return m.initDatabase()
+}
+
+// initDatabase loads tasks from the database into memory.
+func (m *manager) initDatabase() error {
+	return m.db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket([]byte("tasks"))
+		if bucket == nil {
+			return nil
+		}
+
+		return bucket.ForEach(func(k, v []byte) error {
+			var task Task
+			if err := json.Unmarshal(v, &task); err != nil {
+				return err
+			}
+			m.tasks[task.ID] = &task
```

Comment:
Maybe we need to evict the tasks in `StatusProcessing` (residual tasks after acceld crashed) here?